### ISSUE TITLE
Factory Girl should support lazy generation of associations

### ISSUE THREAD

**Issue by ajsharp** [thoughtbot/factory_bot#94]
Are you open to some sort of lazy generation of associations. Ideally, the usage would function something like the following:

```
Factory.define :post do |f|
  f.title 'My Post'
  f.body  '...'
  f.comments { |a| [a.association(:comment)]}
end

post = Factory :post # comments association proxy is still empty, no comments have been created

# now we create the associated comments
post.comments.should_not be_empty # => true
```

I think this would be a major win for FG, because even a moderately complex object graph with a medium-to-large sized test suite quickly becomes slow due to all the DB interaction. Plus, I know some of the newer fixture replacement libraries support this feature (i.e. [fabrication](https://github.com/paulelliott/fabrication)), so it would be great if factory girl had this feature as well ;)

Long story short: if you guys are interested in supporting this feature, I'll take a crack at implementing it.
